mlflow/tracking/client.py

```diff
@@ -818,14 +859,18 @@ def _upload_ended_trace_info(
         Returns:
             The updated TraceInfo object.
         """
-        self._tracking_client.end_trace(
+        return self._tracking_client.end_trace(
```

Comment:
While reviewing, I also noticed that, in `InMemoryTraceClientWithTracking`, we end the trace before uploading the data: https://github.com/mlflow/mlflow/blob/0c7b6322555d3d1a00b2cd6c6310d2f1c23204da/mlflow/tracing/clients/tracking.py#L18-L25. Can we switch the ordering of these operations?

Reply:
Let me check whether the `TraceInfo` update in `self._client._upload_ended_trace_info` call is significant to the trace data upload. I guess it won't but currently the latter uses updated `TraceInfo`.

Reply:
The latter only relies on `mlflow.artifactsLocation` tag from the `TraceInfo`, which should be already set when the trace started. Will reverse the order.